### PR TITLE
Doc for quantization

### PR DESCRIPTION
**Pull request by Cjkkkk**

### CODE REVIEW

File: docs/en_US/Compressor/Overview.md [hunk 1]
```diff
@@ -61,17 +62,34 @@ The function call `pruner.compress()` modifies user defined model (in Tensorflow
...
 * __op_types__: This is to specify what types of operations to be compressed. 'default' means following the algorithm's default setting.
 * __op_names__: This is to specify by name what operations to be compressed. If this field is omitted, operations will not be filtered by it.
 * __exclude__: Default is False. If this field is True, it means the operations with specified types and names will be excluded from the compression.
 
+#### Keys for quantization algorithms
+**If you use quantization algorithms, you need to especify more keys. If you use pruning algorithms, you can safely skip these keys**
+
+* __quant_types__ : list of string. 
+
+Type of quantization you want to apply, currently support 'weight', 'input', 'output'.
```
Comment (QuanluZhang) [inline docs/en_US/Compressor/Overview.md]: is there default value for `quant_types`?

File: docs/en_US/Compressor/Overview.md [hunk 2]
```diff
@@ -61,17 +62,34 @@ The function call `pruner.compress()` modifies user defined model (in Tensorflow
...
+
+* __quant_bits__ : int or dict of {str : int}
+
+bits length of quantization, key is the quantization type, value is the length, eg. {'weight': 8},
+when the type is int, all quantization types share same bits length.
+
+#### Other keys specified for every compression algorithm
 There are also other keys in the `dict`, but they are specific for every compression algorithm. For example, some , some.
```
Comment (QuanluZhang) [inline docs/en_US/Compressor/Overview.md]: could you help update this paragraph?

File: docs/en_US/Compressor/Overview.md [hunk 3]
```diff
@@ -257,7 +273,46 @@ class YourQuantizer(nni.compression.tensorflow.Quantizer):
         """
         pass
 ```
+#### customize backward function
+Sometimes it's necessary for a quantization operation to have a customized backward function, such as Straight-Through Estimator,
```
Comment (QuanluZhang) [inline docs/en_US/Compressor/Overview.md]: please explain `Straight-Through Estimator`, or add reference link for it.

File: docs/en_US/Compressor/Overview.md [hunk 4]
```diff
@@ -257,7 +273,46 @@ class YourQuantizer(nni.compression.tensorflow.Quantizer):
...
+class YourQuantizer(Quantizer):
+    def __init__(self, model, config_list):
+        super().__init__(model, config_list)
+        # set your customized backward function to overwrite default backward function
+        self.quant_grad = ClipGrad
+
+```
 
-__[TODO]__ ...
+The default backward function for quant_weight, quant_input, quant_output is Straight-Through Estimator.
```
Comment (QuanluZhang) [inline docs/en_US/Compressor/Overview.md]: If you do not customize `QuantGrad`, the default backward ...

File: docs/en_US/Compressor/Overview.md (hunk 1 above)
Comment (Cjkkkk) [reply docs/en_US/Compressor/Overview.md]: No.

File: docs/en_US/Compressor/Overview.md (hunk 2 above)
Comment (Cjkkkk) [reply docs/en_US/Compressor/Overview.md]: I add `LevelPruner`'s key `sparsity` as a example.

File: docs/en_US/Compressor/Overview.md (hunk 3 above)
Comment (Cjkkkk) [reply docs/en_US/Compressor/Overview.md]: I add a stackoverflow link to shortly explain this concept. There is also a paper reference link in this answer.

File: docs/en_US/Compressor/Overview.md (hunk 4 above)
Comment (Cjkkkk) [reply docs/en_US/Compressor/Overview.md]: Fixed.

File: docs/en_US/Compressor/Overview.md [hunk 5]
```diff
@@ -61,17 +62,34 @@ The function call `pruner.compress()` modifies user defined model (in Tensorflow
...
+
+#### Common keys
+In each `dict`, there are some keys commonly supported by NNI compression:
 
 * __op_types__: This is to specify what types of operations to be compressed. 'default' means following the algorithm's default setting.
 * __op_names__: This is to specify by name what operations to be compressed. If this field is omitted, operations will not be filtered by it.
 * __exclude__: Default is False. If this field is True, it means the operations with specified types and names will be excluded from the compression.
 
+#### Keys for quantization algorithms
+**If you use quantization algorithms, you need to especify more keys. If you use pruning algorithms, you can safely skip these keys**
```
Comment (chicm-ms) [inline docs/en_US/Compressor/Overview.md]: typo: `especify`

File: docs/en_US/Compressor/Overview.md [hunk 6]
```diff
@@ -61,17 +62,34 @@ The function call `pruner.compress()` modifies user defined model (in Tensorflow
...
+
+* __quant_types__ : list of string. 
+
+Type of quantization you want to apply, currently support 'weight', 'input', 'output'.
+
+* __quant_bits__ : int or dict of {str : int}
+
+bits length of quantization, key is the quantization type, value is the length, eg. {'weight': 8},
+when the type is int, all quantization types share same bits length.
```
Comment (chicm-ms) [inline docs/en_US/Compressor/Overview.md]: `when the type is int` : I don't understand this condition, can you explain more about what `the type` means?

File: docs/en_US/Compressor/Overview.md (hunk 1 above)
Comment (chicm-ms) [reply docs/en_US/Compressor/Overview.md]: Suggest to explain clearly about each of the 3 quant types.

File: docs/en_US/Compressor/Overview.md [hunk 7]
```diff
@@ -30,6 +30,7 @@ We have provided several compression algorithms, including several pruning and q
 | [Naive Quantizer](./Quantizer.md#naive-quantizer) |  Quantize weights to default 8 bits |
 | [QAT Quantizer](./Quantizer.md#qat-quantizer) | Quantization and Training of Neural Networks for Efficient Integer-Arithmetic-Only Inference. [Reference Paper](http://openaccess.thecvf.com/content_cvpr_2018/papers/Jacob_Quantization_and_Training_CVPR_2018_paper.pdf)|
 | [DoReFa Quantizer](./Quantizer.md#dorefa-quantizer) | DoReFa-Net: Training Low Bitwidth Convolutional Neural Networks with Low Bitwidth Gradients. [Reference Paper](https://arxiv.org/abs/1606.06160)|
+| [BNN Quantizer](./Quantizer.md#BNN-Quantizer) | Binarized Neural Networks: Training Deep Neural Networks with Weights and Activations Constrained to +1 or -1. [Reference Paper](https://arxiv.org/abs/1602.02830)|
```
Comment (chicm-ms) [inline docs/en_US/Compressor/Overview.md]: I think it's better to have a brief description of the concept `Quantization` before the table, same for `Pruning`

File: docs/en_US/Compressor/Overview.md (hunk 5 above)
Comment (Cjkkkk) [reply docs/en_US/Compressor/Overview.md]: fix

File: docs/en_US/Compressor/Overview.md (hunk 1 above)
Comment (Cjkkkk) [reply docs/en_US/Compressor/Overview.md]: fix

File: docs/en_US/Compressor/Overview.md (hunk 6 above)
Comment (Cjkkkk) [reply docs/en_US/Compressor/Overview.md]: fix

File: docs/en_US/Compressor/Overview.md [hunk 8]
```diff
@@ -61,17 +62,47 @@ The function call `pruner.compress()` modifies user defined model (in Tensorflow
...
+bits length of quantization, key is the quantization type, value is the quantization bits length, eg. 
+```
+{
+    quant_bits: {
+        'weight': 8,
+        'output': 4,
+        },
+}
+```
+when the key is int type, all quantization types share same bits length. eg.
```
Comment (chicm-ms) [inline docs/en_US/Compressor/Overview.md]: Per the example, it seems the condition should be:  `when the value is int type`

Comment (Cjkkkk) [reply docs/en_US/Compressor/Overview.md]: fixed

File: docs/en_US/Compressor/Overview.md (hunk 7 above)
Comment (Cjkkkk) [reply docs/en_US/Compressor/Overview.md]: I add a reference link to recent survey of model compression and brief introduction of Quantization and Pruning.

**Pull request merged**